assets/js/base/components/cart-checkout/product-sale-badge/index.js

```diff
@@ -17,18 +17,26 @@ import ProductBadge from '../product-badge';
  * @param {Object} props            Incoming props.
  * @param {Object} props.currency   Currency object.
  * @param {number} props.saleAmount Discounted amount.
- *
+ * @param {string} props.format     Format to change the price.
  * @return {*} The component.
  */
-const ProductSaleBadge = ( { currency, saleAmount } ) => {
+const ProductSaleBadge = ( { currency, saleAmount, format = '<price/>' } ) => {
 	if ( ! saleAmount || saleAmount <= 0 ) {
 		return null;
 	}
+	if ( ! format.includes( '<price/>' ) ) {
+		format = '<price/>';
+		// eslint-disable-next-line no-console
+		console.error( 'Price formats need to include the `<price/>` tag.' );
+	}
 	return (
 		<ProductBadge className="wc-block-components-sale-badge">
 			{ createInterpolateElement(
-				/* translators: <price/> will be replaced by the discount amount */
-				__( 'Save <price/>', 'woo-gutenberg-products-block' ),
+				sprintf(
+					/* translators: %s will be replaced by the discount amount */
+					__( `Save %s`, 'woo-gutenberg-products-block' ),
```

Comment:
Do you think it would be worthwhile making the "save" part of this string part of the default format?  I can envision use cases where this may be desirable to change. For example:

Save X

vs

X off

Or even more advance cases such as "buy one get one free" (but this would involve making <price> non-enforced, which may be ok?)